### PR TITLE
fix: use submitted icon on mainnet

### PR DESCRIPTION
**Pull request by just-toby**
## Description

there are two UX changes in this PR (i'll leave comments on the code changes that implement them)
- on mainnet, show a different icon when waiting for transaction confirmation. instead of showing a green check, we show a blue arrow ie the `SubmittedIcon`
- on all chains, remove the `TradeSummary` UI in the "tx confirmed" state. and show the explorer link in its place

there's also a small refactor of the `getContent` function into the `useStepContents` hook - i found a bug in which these contents weren't updating in all expected cases because of the inner rendering function of the `steps` array. so this change is to make all the steps' contents precomputed and memoized.

_Linear ticket:_ https://linear.app/uniswap/issue/WEB-2574/swap-submit-modal-should-have-an-arrow-icon-instead-of-checkmark
_Relevant docs:_ https://www.figma.com/file/ksWAw6EZiiMuRyMQQHicoH/Permit2-Revisions?node-id=1035%3A9685&mode=dev

### Before

current production swap flow:

https://github.com/Uniswap/interface/assets/66155195/ca2bb2e1-bb1a-4ed1-a8e0-f8ab3c2ac3f8

### After

new mainnet UI:

https://github.com/Uniswap/interface/assets/66155195/aeafb71e-bf8c-433f-8c19-ab8a4a6b1940

Explorer link change applies to L2s as well:

https://github.com/Uniswap/interface/assets/66155195/91d0b11b-ae4b-46cd-901a-7b962f274638

### QA (ie manual testing)

- [x] manually verified the animation changes with my eyes and using the devtools feature to slow down animations
- [x] design review

### Automated testing

- [x] Unit test
- [x] Integration/E2E test

### CODE REVIEW

File: src/components/swap/PendingModalContent/index.tsx [hunk 1]
```diff
@@ -282,31 +300,24 @@ export function PendingModalContent({
         )}
         {/* Shown only during the final step under "success" conditions, and scales in. */}
         {currentStep === ConfirmModalState.PENDING_CONFIRMATION && showSuccess && <AnimatedEntranceConfirmationIcon />}
+        {/* Shown only during the final step on mainnet, when the transaction is sent but pending confirmation. */}
+        {currentStep === ConfirmModalState.PENDING_CONFIRMATION && showSubmitted && <AnimatedEntranceSubmittedIcon />}
```
Comment (just-toby) [inline src/components/swap/PendingModalContent/index.tsx:304]: this is the new icon for the mainnet pending state

File: src/components/swap/PendingModalContent/index.tsx [hunk 2]
```diff
@@ -146,70 +145,89 @@ function getContent(args: ContentArgs): PendingModalStep {
...
+          subtitle: showSubmittedState ? explorerLink : tradeSummary,
+          bottomLabel: showSubmittedState ? t`Transaction pending...` : explorerLink,
+        }
+      } else {
+        return {
+          title,
+          subtitle: explorerLink,
+          bottomLabel: null,
+        }
+      }
```
Comment (just-toby) [inline src/components/swap/PendingModalContent/index.tsx:183]: these lines implement the change to remove the `TradeSummary` and replace it with the explorer link. all other changes in this file are just moving code around to make this function a hook

File: src/components/swap/PendingModalContent/Logos.tsx [hunk 3]
```diff
@@ -162,7 +162,36 @@ function ConfirmedIcon({ className }: { className?: string }) {
...
 export const AnimatedEntranceConfirmationIcon = styled(ConfirmedIcon)`
+  position: absolute;
+  height: 48px;
+  width: 48px;
+`
+
+export const AnimatedEntranceSubmittedIcon = styled(SubmittedIcon)`
+  position: absolute;
   height: 48px;
   width: 48px;
```
Comment (cbachmeier) [inline src/components/swap/PendingModalContent/Logos.tsx:196]: can you pull out these duplicate `css` styles to be shared?

File: src/components/swap/PendingModalContent/index.tsx [hunk 4]
```diff
@@ -282,31 +300,24 @@ export function PendingModalContent({
...
+        {((currentStep === ConfirmModalState.PENDING_CONFIRMATION && !showSuccess && !showSubmitted) ||
           tokenApprovalPending ||
           wrapPending ||
           revocationPending) && <LoadingIndicatorOverlay />}
       </LogoContainer>
       <HeaderContainer
         gap="md"
-        $disabled={revocationPending || tokenApprovalPending || wrapPending || (swapPending && !showSuccess)}
+        $disabled={
+          revocationPending || tokenApprovalPending || wrapPending || (swapPending && !showSuccess && !showSubmitted)
```
Comment (cbachmeier) [inline src/components/swap/PendingModalContent/index.tsx:316]: we use the `revocationPending || tokenApprovalPending || wrapPending` condition a couple times, can we pull this out to its own const

File: src/components/swap/PendingModalContent/index.tsx [hunk 5]
```diff
@@ -146,70 +145,89 @@ function getContent(args: ContentArgs): PendingModalStep {
     chainId,
   } = args
 
-  switch (step) {
-    case ConfirmModalState.WRAPPING:
+  const pendingConfirmationContent = useMemo(() => {
```
Comment (cbachmeier) [inline src/components/swap/PendingModalContent/index.tsx:148]: could you move this block to its own function such as `getPendingConfirmationContent` to help maintain the readability of the hook

File: src/components/swap/PendingModalContent/index.tsx [hunk 6]
```diff
@@ -146,70 +145,89 @@ function getContent(args: ContentArgs): PendingModalStep {
...
+    } else if ((swapPending || swapConfirmed) && chainId && swapResult && swapResult.type === TradeFillType.Classic) {
+      const explorerLink = (
+        <ExternalLink
+          href={getExplorerLink(chainId, swapResult.response.hash, ExplorerDataType.TRANSACTION)}
+          color="textSecondary"
+        >
+          <Trans>View on Explorer</Trans>
+        </ExternalLink>
+      )
+      const showSubmittedState = swapPending && chainId === ChainId.MAINNET
```
Comment (cbachmeier) [inline src/components/swap/PendingModalContent/index.tsx:170]: why do we only want to do the submitted state on Mainnet? Speed of L2s? What about other alt L1s?

File: src/components/swap/PendingModalContent/index.tsx [hunk 7]
```diff
@@ -146,70 +145,89 @@ function getContent(args: ContentArgs): PendingModalStep {
...
+      if (swapPending) {
+        return {
+          title,
+          subtitle: showSubmittedState ? explorerLink : tradeSummary,
+          bottomLabel: showSubmittedState ? t`Transaction pending...` : explorerLink,
```
Comment (cbachmeier) [inline src/components/swap/PendingModalContent/index.tsx:175]: we already know `swapPending` is true here so we could just check if current chain is Mainnet

File: src/components/swap/PendingModalContent/index.tsx [hunk 8]
```diff
@@ -146,70 +145,89 @@ function getContent(args: ContentArgs): PendingModalStep {
...
       return {
+        title,
+        subtitle: tradeSummary,
+        bottomLabel: (
+          <ExternalLink href="https://support.uniswap.org/hc/en-us/articles/17515415311501" color="textSecondary">
+            <Trans>Learn more about swapping with UniswapX</Trans>
+          </ExternalLink>
+        ),
+      }
+    } else if ((swapPending || swapConfirmed) && chainId && swapResult && swapResult.type === TradeFillType.Classic) {
```
Comment (cbachmeier) [inline src/components/swap/PendingModalContent/index.tsx:161]: `swapResult && swapResult.type === TradeFillType.Classic` -> `swapResult?.type === TradeFillType.Classic`

File: src/components/swap/PendingModalContent/index.tsx [hunk 9]
```diff
@@ -263,8 +280,9 @@ export function PendingModalContent({
     return <OrderContent order={{ status: order.status, orderHash: order.orderHash, details: order }} />
   }
 
-  // On mainnet, we show the success icon once the tx is sent, since it takes longer to confirm than on L2s.
-  const showSuccess = swapConfirmed || (swapPending && chainId === ChainId.MAINNET)
+  const showSuccess = swapConfirmed
```
Comment (cbachmeier) [inline src/components/swap/PendingModalContent/index.tsx:283]: this `showSuccess` const can just be replaced with `swapConfirmed`

**Review by cbachmeier — CHANGES_REQUESTED** [review]

**Review by cbachmeier — APPROVED** [review]

**Review by JackShort — APPROVED** [review]

**Pull request merged**